pkg/etcd/stream.go

```diff
@@ -42,16 +43,27 @@ func NewStream(logger kitlog.Logger, client etcdGetter, opt StreamOptions) (<-ch
 		defer cancel()
 		defer wg.Done()
 
-		logger.Log("event", "watch.start")
-		for resp := range client.Watch(opt.Ctx, "/", clientv3.WithPrefix()) {
+	Watch:
+
+		logger.Log("event", "watch_start")
+		for resp := range client.Watch(clientv3.WithRequireLeader(ctx), "/", clientv3.WithPrefix()) {
 			for _, event := range resp.Events {
+				if resp.Err() != nil {
+					logger.Log("error", resp.Err(), "msg", "received error from etcd watcher")
+				}
```

Comment:
Do we expect to receive events even if there's an error? If not, I think this wants to sit just above L50?

Reply:
Yes it does 🤦‍♂